src/org/mozilla/javascript/NativeReflect.java

```diff
@@ -0,0 +1,419 @@
+            } else {
+                newTargetPrototype = newTarget.get("prototype", newTarget);
+            }
+
+            if (!(newTargetPrototype instanceof Scriptable)
+                    || ScriptRuntime.isSymbol(newTargetPrototype)
+                    || Undefined.isUndefined(newTargetPrototype)) {
+                newTargetPrototype = null;
+            }
+        }
+
+        // hack to set the right prototype before calling the ctor
+        if (ctor instanceof BaseFunction && newTargetPrototype != null) {
+            BaseFunction ctorBaseFunction = (BaseFunction) ctor;
+            Scriptable result = ctorBaseFunction.createObject(cx, scope);
+            if (result != null) {
+                result.setPrototype((Scriptable) newTargetPrototype);
+
+                Object val = ctorBaseFunction.call(cx, scope, result, callArgs);
+                if (val instanceof Scriptable) {
+                    return (Scriptable) val;
+                }
+
+                return result;
+            }
+        }
+
+        Scriptable newScriptable = ctor.construct(cx, scope, callArgs);
+        if (newTargetPrototype != null) {
+            newScriptable.setPrototype((Scriptable) newTargetPrototype);
+        }
+
+        return newScriptable;
+    }
+
+    private static Object defineProperty(
+            Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        if (args.length < 3) {
+            throw ScriptRuntime.typeErrorById(
+                    "msg.method.missing.parameter",
+                    "Reflect.defineProperty",
+                    "3",
+                    Integer.toString(args.length));
+        }
+
+        ScriptableObject target = checkTarget(args);
+        ScriptableObject desc = ScriptableObject.ensureScriptableObject(args[2]);
+
+        try {
+            return target.defineOwnProperty(cx, args[1], desc);
+        } catch (EcmaError e) {
+            return false;
+        }
+    }
+
+    private static Object deleteProperty(
+            Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        ScriptableObject target = checkTarget(args);
+
+        if (args.length > 1) {
+            if (ScriptRuntime.isSymbol(args[1])) {
+                return ScriptableObject.deleteProperty(target, (Symbol) args[1]);
+            }
+            return ScriptableObject.deleteProperty(target, ScriptRuntime.toString(args[1]));
+        }
+
+        return false;
+    }
+
+    private static Object get(Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        ScriptableObject target = checkTarget(args);
+
+        if (args.length > 1) {
+            if (ScriptRuntime.isSymbol(args[1])) {
+                Object prop = ScriptableObject.getProperty(target, (Symbol) args[1]);
+                return prop == Scriptable.NOT_FOUND ? Undefined.SCRIPTABLE_UNDEFINED : prop;
+            }
+            if (args[1] instanceof Double) {
```

Comment:
Think you already mentioned this elsewhere, but likely should check for Number, not Double

Reply:
fixed in the new pr